### PR TITLE
[CORDA-3442] Update app upgrade notes for gradle version for PV5

### PR DESCRIPTION
**Pull request by JamesHR3**
This PR updates the app upgrade notes to indicate that the gradle version has changed, and that an additional repository must be added to the gradle file to pick up the gradle-api-tooling dependency.

### CODE REVIEW

File: docs/source/app-upgrade-notes.rst [hunk 1]
```diff
@@ -85,6 +85,24 @@ To fix this, an explicit type hint must be provided to the compiler:
...
 This stops type inference from occurring and forces the variable to be of type ``AbstractParty``.
 
+Step 2. Update Gradle version and associated dependencies
+---------------------------------------------------------
+
+Platform Version 5 requires Gradle 5.4 to build. If you use the Gradle wrapper, you can upgrade by running:
+
+ .. code:: shell
+
+        ./gradlew wrapper --gradle-version 5.4.1
```
Comment (anthonykeenan) [inline docs/source/app-upgrade-notes.rst:97]: Oooh I didn't know you could do that I've just been editing it manually.

File: docs/source/app-upgrade-notes.rst [hunk 2]
```diff
@@ -85,6 +85,24 @@ To fix this, an explicit type hint must be provided to the compiler:
...
+
+Otherwise, upgrade your installed copy in the usual manner for your operating system.
+
+Additionally, you'll need to add https://repo.gradle.org/gradle/libs-releases as a repository to your project, in order to pick up the
+gradle-api-tooling dependency. You can do this by adding the following to the repositories in your Gradle file:
```
Comment (anthonykeenan) [inline docs/source/app-upgrade-notes.rst]: I think gradle-api-tooling needs backticks

File: docs/source/app-upgrade-notes.rst [hunk 3]
```diff
@@ -124,6 +142,10 @@ You should also ensure you're using Gradle 4.10 (but not 5). If you use the Grad
 
 Otherwise just upgrade your installed copy in the usual manner for your operating system.
 
+.. note:: Platform Version 5 requires a different version of Gradle, so if you're intending to upgrade past Platform Version 4 you may wish
+   to skip updating Gradle here and upgrade directly to the version required by Platform Version 5. You'll still need to alter the version
+   numbers in your Gradle file as shown in this section. See `Step 2. Update Gradle version and associated dependencies`_
```
Comment (anthonykeenan) [inline docs/source/app-upgrade-notes.rst]: Have you build the docs and checked whether that link actually works? I thought you had to do something funny with a `::ref` link

**Review by anthonykeenan — CHANGES_REQUESTED** [review]
Couple of minor things

File: docs/source/app-upgrade-notes.rst (hunk 1 above)
Comment (JamesHR3) [reply docs/source/app-upgrade-notes.rst:97]: We suggest doing the same thing in the 4.1 upgrade notes :)

File: docs/source/app-upgrade-notes.rst (hunk 3 above)
Comment (JamesHR3) [reply docs/source/app-upgrade-notes.rst]: I had built locally and this did work for me (and is part of the rst spec as far as I can tell), however we consistently don't do this elsewhere, so I've switched to an explicit `:ref:` to match everywhere else in the docs.

**Review by anthonykeenan — APPROVED** [review]

**Pull request merged**